### PR TITLE
[14.0][ADD] product_state_active

### PR DESCRIPTION
**Pull request by rousseldenis**
Depends on :

- [ ] #947

### CODE REVIEW

**Comment by rousseldenis** [comment]
@simahawk @dreispt This is quite simple and avoid to have such things:

https://github.com/OCA/product-attribute/blob/14.0/product_tier_validation/models/product_template.py#L19

The tier validation IMHO should only do its stuff: doing validation. In combination with this, you can define the state that will trigger the active change.

The point is: that's the state change that should only trigger the active change, not the validation.

What do you think about ?

File: product_state_active/models/product_state.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductState(models.Model):
+
+    _inherit = "product.state"
+
+    deactivate_product = fields.Boolean(
+        help="Check this if you want to inactive product when reaching this state."
```
Comment (simahawk) [inline product_state_active/models/product_state.py]:
```suggestion
        help="Check this if you want to archive products when reaching this state."
```

File: product_state_active/models/product_state.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    )
+    activate_product = fields.Boolean(
+        help="Check this if you want to active product when reaching this state."
```
Comment (simahawk) [inline product_state_active/models/product_state.py]:
```suggestion
        help="Check this if you want to un-archive products when reaching this state."
```

**Review by simahawk — APPROVED** [review]

File: product_state_active/models/product_state.py (hunk 2 above)
Comment (rousseldenis) [reply product_state_active/models/product_state.py]: @simahawk Done. Please review too #947

**Review by flotho — APPROVED** [review]
LGTM

**Pull request merged**